idb/postgres/postgres.go

```diff
@@ -85,6 +86,30 @@ func openPostgres(db *sql.DB, opts *idb.IndexerDbOptions, logger *log.Logger) (p
 	return
 }
 
+func txnWithRetry(f func() error) error {
+	maxDelay := time.Minute
+	delay := time.Millisecond
+
+	for {
+		err := f()
+
+		// If not serialization error.
+		var pqerr *pq.Error
+		if !errors.As(err, &pqerr) || (pqerr.Code != "40001") {
+			return err
+		}
+
+		fmt.Println("retrying transaction")
+		// Randomize delay to avoid conflicts.
+		time.Sleep(time.Duration(rand.Int63n(int64(delay))))
```

Comment:
Not sure we need a delay, do both transactions fail? I would think the other had to succeed in order for this one to fail.